lib/Dialect/Arc/Transforms/GroupResetsAndEnables.cpp

```diff
@@ -166,7 +165,8 @@ struct GroupAssignmentsInIfPattern : public OpRewritePattern<scf::IfOp> {
     // Skip anything not in a ClockTreeOp
     auto clockTreeOp = ifOp->getParentOfType<ClockTreeOp>();
     if (!clockTreeOp)
-      return failure();
+      // This probably means the design uses a derived clock (warning).
+      return success(false);
```

Comment:
This is exactly equivalent to `return failure()` (see implementation in LogicalResult.h), so no need to change.

Reply:
Oh thanks, I ran into an issue where the compiler was failing without displaying an error during debugging and thought it was because of this, but it must have been something else related to the dominance stuff. I undid that change, thanks for the catch!

Reply:
I see. That might be because no failure is returned in line 207 (also no error message). Thanks for the catch!

Reply:
Should be fixed in 227ae284a9c0d8353307b5e1125cf34571fd7e87